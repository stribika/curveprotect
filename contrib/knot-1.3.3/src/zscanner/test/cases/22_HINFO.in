$ORIGIN	.
$TTL	1

; OK
@	HINFO	"" ""				; Blank string
@	HINFO	a b				; One char string
@	HINFO	\  \ 				; One space char
@	HINFO	"\ " "\ "			; One space char in quotes
@	HINFO	\021 \022			; One unprintable char
@	HINFO	"\\ \""	"\\ \""			; Special chars
@	HINFO	first \#			; Array with special string
@	HINFO	\# 2 00	00			; Hexadecimal rdata
@	TYPE13	\# 2 00	00			; TYPE + Hexadecimal rdata
@	TYPE13	"" ""				; TYPE
@	HINFO	"abcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMN1234\053" ""	; Text string of maximal length (255 chars)
@	hinfo	"" ""				; Type in lower-case

; KO
@	HINFO
@	HINFO			; Empty rdata
@	HINFO	\# 0		; Hex empty rdata
@	HINFO	\#		; Missing hex length
@	HINFO	\# second	; Hex notation with second text string
@	HINFO	\01 ""		; Missing digit in decimal notation
@	HINFO	\256 ""		; 8bit overflow in decimal notation
@	HINFO	\# 3 0000	; Bad rdata length
@	HINFO	""" ""		; '"' char without forward slash
@	HINFO	"" "" ""	; Unexpected item
@	HINFO	"abcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNabcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMN1234\0536" ""	; Maximal length overflow
