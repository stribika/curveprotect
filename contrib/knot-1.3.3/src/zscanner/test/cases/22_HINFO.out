OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=000D
RDATA=0000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=000D
RDATA=01610162
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=000D
RDATA=01200120
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=000D
RDATA=01200120
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=000D
RDATA=01150116
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=000D
RDATA=035C2022035C2022
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=000D
RDATA=0566697273740123
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=000D
RDATA=0000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=000D
RDATA=0000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=000D
RDATA=0000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=000D
RDATA=FF6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E6162636465666768696A6B6C6D6E6F707172737475767778797A313233343536373839304142434445464748494A4B4C4D4E313233343500
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=000D
RDATA=0000
------
WARNG=ZSCANNER_EBAD_RDATA
------
WARNG=ZSCANNER_EBAD_RDATA
------
WARNG=ZSCANNER_EBAD_HEX_CHAR
------
WARNG=ZSCANNER_EBAD_NUMBER
------
WARNG=ZSCANNER_EBAD_NUMBER
------
WARNG=ZSCANNER_EBAD_NUMBER
------
WARNG=ZSCANNER_ENUMBER8_OVERFLOW
------
WARNG=ZSCANNER_EBAD_RDATA_LENGTH
------
WARNG=ZSCANNER_EBAD_TEXT
------
WARNG=ZSCANNER_EBAD_REST
------
WARNG=ZSCANNER_EITEM_OVERFLOW
------
